src/codec/mod.rs

```diff
@@ -248,6 +275,17 @@ fn decode_request_pdu_bytes(bytes: &Bytes) -> io::Result<Request<'static>> {
             }
             ReadWriteMultipleRegisters(read_address, read_quantity, write_address, data.into())
         }
+        0x2b if rdr.read_u8()? == MEI_TYPE_READ_DEVICE_IDENTIFICATION => {
```

Comment:
Using a side-effect within a match guard is both confusing and an error. The value is consumed and lost for the following match arms.